webapp/src/app/components/load-map/virtualMapNode.model.ts

```diff
@@ -0,0 +1,48 @@
+import { MapNode } from './mapNode.model';
+
+/**
+ * A class that only returns the children that are included in the filter.
+ */
+export class VirtualMapNode {
+    private original: MapNode;
+    private knownChildren = new WeakMap<MapNode, VirtualMapNode>();
+    
+    public constructor(original: MapNode) {
+        this.original = original;
+
+        if (original.children) {
+            for (const node of original.children) {
+                this.knownChildren.set(node, new VirtualMapNode(node));
+            }
+        }
+    }
+
+    public get name(): string {
+        return this.original.name;
+    }
+
+    public get path(): string | undefined {
+        return this.original.path;
+    }
+
+    public get children(): VirtualMapNode[] | undefined {
+        if (!this.original.children) {
+            return undefined;
+        }
+
+        return this.original.children
+            .filter((n): boolean => n.displayed)
```

Comment:
is "boolean" really needed? `.filter(n => n.displayed)` should work

Reply:
I could have sworn the linter requires these but it seems I was wrong. Fixed.